src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -62,34 +62,87 @@
 
 // Consider moving utility classes to their own file
 #pragma mark - Utility Classes
-// This class is for storing weak references in a container
-@interface MTRWeakReference<ObjectType> : NSObject
-+ (instancetype)weakReferenceWithObject:(ObjectType)object;
-- (instancetype)initWithObject:(ObjectType)object;
-- (ObjectType)strongObject; // returns strong object or NULL
-@end
 
-@interface MTRWeakReference () {
+// container of MTRDevice delegate weak reference, its queue, and its interested paths for attribute reports
+MTR_DIRECT_MEMBERS
+@interface MTRDeviceDelegateInfo : NSObject {
 @private
-    __weak id _object;
+    void * _delegatePointerValue;
+    __weak id _delegate;
+    dispatch_queue_t _queue;
+    NSArray * _interestedPathsForAttributes;
```

Comment:
Should be marked nullable?

Reply:
We don't seem to consistently do this in `.mm` files currently, but we should (including `NS_ASSUME_NONNULL...` etc)